utils.js

```diff
@@ -0,0 +1,155 @@
+    $('#daily-' + i).append(daily);
+  };
+};
+
+//posts error message for inputted city that does not exist
+var postError = function(query, errorMsg) {
+  $('#weather-info').empty();
+  $('#weather-info-img').empty();
+  $('.daily').remove();
+
+  $('#weather-info').append('<div>' + query + ' ' + errorMsg + '</div>');
+}
+
+//retrieves current weather information
+var fetchCurrent = function (query) {
+  $.ajax({
+    method: "GET",
+    url: "http://api.openweathermap.org/data/2.5/weather?q=" + query + ",US&units=imperial&APPID=9c1a2a4005cf6abf4da3b1086d6c56e8",
+    dataType: "json",
+    success: function(data) {
+      postCurrent(data);
+    },
+    error: function(jqXHR, textStatus, errorThrown) {
+      postError(query, errorThrown);
+    }
+  });
+};
+
+//retrieves weather forecast information
+var fetchForecast = function (query) {
+  $.ajax({
+    method: "GET",
+    url: 'http://api.openweathermap.org/data/2.5/forecast?q=' + query + ',US&units=imperial&APPID=9c1a2a4005cf6abf4da3b1086d6c56e8',
+    dataType: "json",
+    success: function(data) {
+      postForecast(data);
+    },
+    error: function(jqXHR, textStatus, errorThrown) {
+      postError(query, errorThrown);
+    }
+  });
+};
+
+//retrieve current weather based off of user current location
+var fetchCurrentGeo = function(lat, long) {
+  $.ajax({
+    method: "GET",
+    url: 'http://api.openweathermap.org/data/2.5/weather?lat=' + lat + '&lon=' + long + '&units=imperial&APPID=9c1a2a4005cf6abf4da3b1086d6c56e8',
+    dataType: "json",
+    success: function(data) {
+      postCurrent(data);
+    },
+    error: function(jqXHR, textStatus, errorThrown) {
+      postError('Location ' + errorThrown)
+    }
+  });
+};
+
+//retrieve weather forecast based off of user current location
+var fetchForecastGeo = function(lat, long) {
+  $.ajax({
+    method: "GET",
+    url: "http://api.openweathermap.org/data/2.5/forecast?lat=" + lat + "&lon=" + long + "&units=imperial&APPID=9c1a2a4005cf6abf4da3b1086d6c56e8",
+    dataType: "json",
+    success: function(data) {
+      postForecast(data);
+    },
+    error: function(jqXHR, textStatus, errorThrown) {
+      postError('Location ' + errorThrown)
+    }
+  });
+};
+
+//retrieve user current location
+var findLocation = function() {
+  navigator.geolocation.getCurrentPosition(function(position) {
+    fetchCurrentGeo(position.coords.latitude, position.coords.longitude);
+    fetchForecastGeo(position.coords.latitude, position.coords.longitude);
+  });
+};
+
+//sets current city as default city
+var setDefault = function (data) {
+
+  if (data && localStorage.getItem('default') != data.name) {
+    localStorage.setItem('default', data.name);
+  };
+}
+
+var getDefault = function() {
+  if (localStorage.getItem('default')) {
+    fetchCurrent(localStorage.getItem('default'));
```

Comment:
Instead of having 4 "fetch" functions, you could have narrowed this down to at least two, maybe even one. Really the only thing that's changing URL and which "render" method you're calling on `success`